components/ai_chat/resources/untrusted_conversation_frame/untrusted_conversation_frame.tsx

```diff
@@ -10,12 +10,70 @@ import '$web-components/app.global.scss'
 import '$web-common/defaultTrustedTypesPolicy'
 import ConversationEntries from './components/conversation_entries'
 import { UntrustedConversationContextProvider } from './untrusted_conversation_context'
+import UntrustedConversationFrameAPI from './untrusted_conversation_frame_api'
 
 import '../common/strings'
 
 setIconBasePath('chrome-untrusted://resources/brave-icons')
 
 function App() {
+  // Iframe drag detection with proper cleanup
+  React.useEffect(() => {
+    let dragCounter = 0
+    const api = UntrustedConversationFrameAPI.getInstance()
+
+    const handleDragEnter = (e: DragEvent) => {
+      if (e.dataTransfer?.types?.includes('Files')) {
+        dragCounter++
+        if (dragCounter === 1) {
+          api.parentUIFrame.dragStart()
+        }
+      }
+    }
+
+    const handleDragLeave = (e: DragEvent) => {
+      if (e.dataTransfer?.types?.includes('Files')) {
+        dragCounter--
+        if (dragCounter === 0) {
+          api.parentUIFrame.dragEnd()
+        }
+      }
+    }
+
+    const handleDragEnd = (e: DragEvent) => {
+      dragCounter = 0
+      api.parentUIFrame.dragEnd()
+    }
+
+    const handleDragOver = (e: DragEvent) => {
+      e.preventDefault()
+    }
+
+    const handleDrop = (e: DragEvent) => {
+      // Don't prevent default - let event bubble to parent overlay
+      // for file processing
+      dragCounter = 0
+      api.parentUIFrame.dragEnd()
+    }
```

Comment:
does this get fired? We display an overlay over the iframe when a drag starts, so I would expect all the events to get processed there (but my brain gets broken by drag and drop on its own, let alone introducing cross iframe communication).